distributed/worker.py

```diff
@@ -2759,18 +2767,37 @@ def get_client(address=None, timeout=None, resolve_address=True) -> Client:
         pass
     else:
         if not address or worker.scheduler.address == address:
-            return worker._get_client(timeout=timeout)
-
-    from distributed.client import Client
+            client = worker._get_client(timeout=timeout)
+            if asynchronous is None:
+                return client
```

Comment:
```suggestion
            if asynchronous is None or client.asynchronous == asynchronous:
                return client
```

Reply:
Could be mistaken, but doesn't seem we can use `asynchronous` property directly, as it is biased for checking if it's in an async call, so went for `_asynchronous` property, reflecting what the client was instantiated with regardless of current call context. https://github.com/dask/distributed/pull/7871/commits/224cae77c38ab495a316fd63f9be10f0c31fc960